### PR TITLE
Elo history chart

### PR DESCRIPTION
**Pull request by AkkarinForest**

### CODE REVIEW

File: app/services/elo/history_generator.rb [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+    def next_ranking(match)
+      next_ranking_calculator.call(current_ranking, match)
+    end
+
+    def current_ranking
+      if history.present?
+        history.newest_ranking
+      else
+        initial_ranking
+      end
```
Comment (AkkarinForest) [inline app/services/elo/history_generator.rb]: maybe history object should be responsible for this logic? I mean `history.newest_ranking` should return a new ranking or if there is none then initial_ranking. Then InitialRanking class would also move there. But I am not sure if history should know about the logic that assigns points to new players.

File: app/services/player_page/elo_points_history_generator.rb [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+      elo_rankings.each do |date, ranking|
+        points = points_for_player(player, ranking)
+
+        elo_points_history.add_points_entry(date, points)
+      end
+    end
+
+    def points_for_player(player, ranking)
+      ranking[player]
```
Comment (AkkarinForest) [inline app/services/player_page/elo_points_history_generator.rb]: This looks awkward, points_for_player shouldn't take ranking? I guess ranking rows could be objects (not a hash) and they should have a method points_for(player)?

File: spec/services/player_page/elo_points_history_generator_spec.rb [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+    expect(subject.call(mark)).to eq({})
+  end
+
+  it "returns player history of elo points for each days they played" do
+    (0..10).step(5) do |number|
+      create(
+        :match,
+        player_1: mark,
+        first_team_score:  '2',
+        second_team_score: '0',
```
Comment (shadre) [inline spec/services/player_page/elo_points_history_generator_spec.rb]: Can we omit scores here?

File: app/services/elo/history_generator.rb [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+    def initial_ranking
+      initial_ranking_creator.call
+    end
+
+    def matches
+      matches_repository.fetch_matches
+    end
+  end
+
+  class InitialRankingCreator
```
Comment (shadre) [inline app/services/elo/history_generator.rb]: I would move it to a separate file and consider using it in both places where this logic occurs, as discussed before.

**Pull request merged**